From ce50e80781edb055ca94a8421dd71440fd22bf73 Mon Sep 17 00:00:00 2001
From: Quanyang Wang <quanyang.wang@windriver.com>
Date: Wed, 2 Sep 2015 06:21:05 +0000
Subject: [PATCH 1/3] kexec: lsi-acp34xx: clear PMU register in secondary
 kernel

When kexec/kdump boot the second kernel, the value in register PMRN_PMUGC0
is not zero but 0x2, at this moment initialize this register may hang the
system, so clear the register first.

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 arch/powerpc/perf/core-lsi-acp.c |   10 +++++++++-
 1 files changed, 9 insertions(+), 1 deletions(-)

diff --git a/arch/powerpc/perf/core-lsi-acp.c b/arch/powerpc/perf/core-lsi-acp.c
index d9dd4a3..47c2cc6 100644
--- a/arch/powerpc/perf/core-lsi-acp.c
+++ b/arch/powerpc/perf/core-lsi-acp.c
@@ -703,7 +703,15 @@ int register_acp_pmu(struct acp_pmu *pmu)
 
 	for_each_possible_cpu(core) {
 		u32 pmgc0;
-
+#ifdef CONFIG_CRASH_DUMP
+		/* when kexec/kdump boot the second kernel, the value in register
+		 * PMRN_PMUGC0 is not zero but 0x2, at this moment initialize this
+		 * register may hang the system, so clear the register first.
+		 */
+		mtdcrx(PMUDCRAI(core), PMRN_PMUGC0);
+		pmgc0 = PMUGC0_PMCC;
+		mtdcrx(PMUDCRDI(core), pmgc0);
+#endif
 		mtdcrx(PMUDCRAI(core), PMRN_PMUGC0);
 		pmgc0 = mfdcrx(PMUDCRDI(core));
 		pmgc0 |= (PMUGC0_LFAC|PMUGC0_PMCC);
-- 
1.7.5.4

